Zend/zend_inheritance.c

```diff
@@ -1700,9 +1783,14 @@ static void zend_do_traits_property_binding(zend_class_entry *ce, zend_class_ent
 
 			Z_TRY_ADDREF_P(prop_value);
 			doc_comment = property_info->doc_comment ? zend_string_copy(property_info->doc_comment) : NULL;
-			zend_declare_property_ex(ce, prop_name,
-									 prop_value, flags,
-								     doc_comment);
+			if (property_info->type) {
+				zend_declare_typed_property(ce, prop_name, prop_value, flags, doc_comment,
```

Comment:
Maybe we should accept a zend_type in declare_typed_property? Would also be more forward compatible with future type-system extensions...